pkg/apis/pipeline/v1/param_types.go

```diff
@@ -505,12 +517,9 @@ func ArrayReference(a string) string {
 
 // validatePipelineParametersVariablesInTaskParameters validates param value that
 // may contain the reference(s) to other params to make sure those references are used appropriately.
-func validatePipelineParametersVariablesInTaskParameters(params []Param, prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
-	taskParamNames := sets.NewString()
-	for i, param := range params {
-		if taskParamNames.Has(param.Name) {
-			errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("params names must be unique, the same param: %s is defined multiple times at", param.Name), fmt.Sprintf("params[%d].name", i)))
-		}
+func validatePipelineParametersVariablesInTaskParameters(params Params, prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
+	errs = errs.Also(params.validateDuplicateParameters().ViaField("params"))
```

Comment:
maybe we don't need the viaField because it's already accounted for in validateDuplicateParameters function?
```suggestion
	errs = errs.Also(params.validateDuplicateParameters()
```